Lib/tempfile.py

```diff
@@ -130,6 +130,11 @@ def _sanitize_params(prefix, suffix, dir):
             dir = gettempdir()
         else:
             dir = gettempdirb()
+    if isinstance(dir, _os.PathLike):
```

Comment:
We should fist checkout the output type, to respect other arguments (prefix and suffix), then decide how to convert dir when it is a path-like objects.